bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableConfiguration.java

```diff
@@ -43,54 +43,54 @@ public class BigtableConfiguration {
   @InternalApi("For internal usage only")
   public static final String HBASE_CLIENT_ASYNC_REGISTRY_IMPL = "hbase.client.registry.impl";
 
-  private static final String[] CONNECTION_CLASS_NAMES = {
-    "com.google.cloud.bigtable.hbase1_x.BigtableConnection",
-    "com.google.cloud.bigtable.hbase2_x.BigtableConnection",
-  };
-
-  private static final Class<? extends Connection> CONNECTION_CLASS = chooseConnectionClass();
-
-  @SuppressWarnings("unchecked")
-  private static Class<? extends Connection> chooseConnectionClass() {
-    for (String className : CONNECTION_CLASS_NAMES) {
-      try {
-        return (Class<? extends Connection>) Class.forName(className);
-      } catch (ClassNotFoundException ignored) {
-        // This class is not on the classpath, so move on to the next className.
-      }
-    }
-    return null;
-  }
-
   /**
    * For internal use only - public for technical reasons.
    *
    * @return the default bigtable {@link Connection} implementation class found in the classpath.
    */
+  @SuppressWarnings("unchecked")
   @InternalApi("For internal usage only")
   public static Class<? extends Connection> getConnectionClass() {
-    Preconditions.checkState(
-        CONNECTION_CLASS != null,
-        "Could not load a concrete implementation of BigtableTableConnection: "
-            + "failed to find bigtable-hbase-1.x on the classpath.");
-    return CONNECTION_CLASS;
-  }
+    String hbaseVersion = VersionInfo.getVersion();
 
-  private static Class<?> getConnectionRegistryClass() {
+    final String connectionClassName;
+
+    if (VersionInfo.compareVersion(hbaseVersion, "2.0") >= 0) {
+      connectionClassName = "com.google.cloud.bigtable.hbase2_x.BigtableConnection";
```

Comment:
please add some tests for this

Reply:
This is already covered by the existing ConnectionTests:
https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/test/java/com/google/cloud/bigtable/hbase/TestBigtableConnection.java

https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/test/java/com/google/cloud/bigtable/hbase/TestBigtableConnection.java